Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,453 @@
+                                              DeviceEventCategory == 2097152, "Expanded - WLAN IDS Activity (2097152)",
+                                              DeviceEventCategory == 4194304, "Expanded - SonicPoint Activity (4194304)",
+                                              DeviceEventCategory == 8388608, "Expanded - Unified Policy Engine (8388608)",
+                                              "Legacy Category Not Mapped"
+                                              )
+  | extend sosIPSPriority = case(ipspri == 1, "High (1)",
+                                              ipspri == 2, "Medium (2)",
+                                              ipspri == 3, "Low (3)",
+                                              ""
+                                              )
+  | extend sosAntiSpywarePriority = case(spypri == 1, "High (1)",
+                                              spypri == 2, "Medium (2)",
+                                              spypri == 3, "Low (3)",
+                                              ""
+                                              )
+  | extend
+      EventVendor = "SonicWall"
+    , EventProduct = "Firewall"
+    , sosDeviceModel = DeviceProduct
+    , DvcOs = "SonicOS"
+    , DvcOsVersion = EventProductVersion
+    , DvcIdType = "Other"
+    , Dvc = sosSerialNumber
+    , DvcDescription = DeviceProduct
+    , ASimMatchingHostname = "-"
+    , ASimMatchingIpAddr = "-"
+    , NetworkIcmpType = tostring(NetworkIcmpType)
+    , NetworkIcmpCode = toint(NetworkIcmpCode)
+    , NetworkBytes = tolong(coalesce(toint(ReceivedBytes), 0) + coalesce(toint(SentBytes), 0))
+    , FieldDeviceCustomNumber1Label = column_ifexists("FieldDeviceCustomNumber1Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , FieldDeviceCustomNumber2Label = column_ifexists("FieldDeviceCustomNumber2Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber1Label = column_ifexists("DeviceCustomNumber1Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber2Label = column_ifexists("DeviceCustomNumber2Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber3Label = column_ifexists("DeviceCustomNumber3Label", "") // Connection Duration in milliseconds (for Connnection Closed msgs)
+    , fw_action = column_ifexists("fw_action", "")
+    , sosFirewallAction = column_ifexists("fw_action", "")
+    , susr = column_ifexists("susr", "") // Logged-in username associated with the log event.
+    , af_polid = column_ifexists("af_polid", "")
+    , af_action = column_ifexists("af_action", "")
+    , af_policy = column_ifexists("af_policy", "")
+    , af_service = column_ifexists("af_service", "")
+    , af_type = column_ifexists("af_type", "")
+    , af_object = column_ifexists("af_object", "")
+    , contentObject = column_ifexists("contentObject", "") // App Firewall/App Rule Content.
+    , srcV6 = column_ifexists("srcV6", "") // Source IPv6 IP

```

Comment:
not found in parse-kv step. Is this part of additionalextension?

Reply:
Fixed this. Included in parse-kv and added a coalesce() to catch either the v4 or v6 address.

Reply:
@jaimeesc I'm re-opening all last comments. Please re-check.